components/form/overview.md

```diff
@@ -0,0 +1,199 @@
+---
+title: Overview
+page_title: Form Overview
+description: Overview of the Form for Blazor.
+slug: form-overview
+tags: telerik,blazor,form,edit,form
+published: True
+position: 0
+---
+
+# Form Overview
+
+The Form for Blazor allows you to generate a form based on your model and to manage customized forms. You can control the component through various parameters, achieve the desired layout by using the [default editor](#automatic-generation-of-fields) or add [custom ones]({%slug form-formitems%}), set the [orientation]({%slug form-orientation%}) and organize those editors in [groups]({%slug form-formgroups%}) and [columns]({%slug form-columns%}). 
+
+#### This article is separated in the following sections:
+
+* [Use the Telerik Form for Blazor with a model](#use-the-telerik-form-for-blazor-with-a-model)
+* [Use the Telerik Form for Blazor with an EditContext](#use-the-telerik-form-for-blazor-with-an-editcontext)
+* [Automatic Generation of fields](#automatic-generation-of-fields)
+* [Data Annotation Attributes](#data-annotation-attributes)
+* [Features](#features)
+
+
+## Use the Telerik Form for Blazor with a model
+
+To use the Form component with a model: 
+
+1. Add the `<TelerikForm>` tag.
+
+1. Provide an object to the `Model` parameter of the component. 
+
+1. Use the `<FormValidation>` tag and in it, provide a validator - like the `DataAnnotationsValidator` that comes with the framework, to enable form validation. 
+
+
+````CSHTML
+@* Provide a model to the Telerik Form *@
+
+@using System.ComponentModel.DataAnnotations
+
+<TelerikForm Model="@person">
+    <FormValidation>
+        <DataAnnotationsValidator />
+    </FormValidation>
+</TelerikForm>
+
+@code {
+    public Person person = new Person();
+
+    public class Person
+    {
+        [Range(100, 1000, ErrorMessage ="The Id must be between 100 and 1000")]
+        public int Id { get; set; }
+        [Required]
+        [MaxLength(20, ErrorMessage ="The first name should be maximum 20 characters long")]
+        public string FirstName { get; set; }
+        [Required]
+        [MaxLength(25, ErrorMessage = "The last name should be maximum 25 characters long")]
+        public string LastName { get; set; }
+        [Required]
+        public DateTime? DOB { get; set; }
+    }
+}
+````
+
+## Use the Telerik Form for Blazor with an EditContext
+
+The Telerik Form for Blazor can utilize the <a href="https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.forms.editcontext?view=aspnetcore-5.0">EditContext class</a>. You can use the events and methods provided by the EditContext to provide custom business logic. 
+
+To use the Form component with an EditContext: 
+
+1. Add the `<TelerikForm>` tag.
+
+1. Provide an object of type `EditContext` to the `EditContext` parameter of the Form. 
+
+1. Use the `<FormValidation>` tag and provide a validator in it - like the `DataAnnotationsValidator` that comes with the framework, to enable form validation. 
```

Comment:
Perhaps, it will be cool to add the alternative to use the extension method as well:

```
<TelerikForm EditContext="@MyEditContext">
</TelerikForm>
@code {
    public EditContext MyEditContext { get; set; }
    public Person person = new Person();
    protected override void OnInitialized()
    {
        MyEditContext = new EditContext(person);
        MyEditContext.AddDataAnnotationsValidation()
    }
```

Reply:
I don't think we should do that in our overview, I think people who want to use the EditContext should be familiar with its features and capabilities. We shold not try to compete with msdn

Reply:
I think it's pivotal to use the `<FormValidation>` since it is connected to our component. This is a great alternative, but it is not connected to our Form. What do you think?

Reply:
I am marking this as resolved. Let's see feedback and add such complications as needed and if people ask.